### DISCUSSION TITLE
[mobile] mutation 발생 시 로딩 추가에 대한 의견

### DISCUSSION THREAD

**Discussion by kingyong9169** [CMI-OSS/cbnu-alrami#848] [Ideas]
mutation이 발생할 때 즉, 구독, 북마크 등등 서버에 post, patch 등 통신을 요청할 때, 로딩 시간 동안 앱의 반응이 없어 UX에 좋지 않을 것 같다는 생각입니다. 작업량이 많으니 릴리즈 이후에 리팩토링 아이디어에 넣어서 어떻게 할지 UX를 설계해보면 좋을 것 같아요! @CMI-OSS/cbnu_alrami_fe

**Comment by nohgijin** [comment]
서버환경이 바뀌면서 현재 UX에 해가될 만큼 반응이 없는건 아닌 것 같은데, 혹시 바뀌기 전에 생각하신 아이디어이신걸까용?

**Reply by kingyong9169** [reply]
아뇨아뇨 저는 서버 환경에 상관없이 사용자한테 로딩 중이라는걸 보여주는게 좋은 것 같아요. 페북이나 인스타는 사용자가 누르는 즉시 반응이 오지만(아마 어떤 처리를 해주지 않았을까 싶습니다.) 저희의 경우에는 현재 빠른 환경에서도 조금의 딜레이는 있고 서버가 갑자기 느려질 수도 문제가 생길수도 있어서 처리해줄 필요가 있다고 생각합니다.
저희 코드를 보면 mutation 로딩, 에러에 대한 처리가 없어서 이 부분은 필요할 것 같아요. UX에 대한 정답은 없어서 고민은 해봐야 할 것 같아요. 다만 제 아이디어를 말씀드리자면,

로딩: 버튼에 해당하는 부분만 로딩 스피너, UI는 회색으로 색상만 바꾸고 버튼 disabled 처리, 페이지 정중앙에 로딩 스피너 띄우기(기존 UI들은 그대로 보여주며 다른 행동은 못하게 막음)
에러: 유지보수하기 쉽도록 서버 및 커스텀 에러 메시지 띄우기

**Reply by kingyong9169** [reply]
react query에 onMutate옵션 사용하면 될 것 같습니다!
